silkworm/capi/silkworm_test.cpp

```diff
@@ -808,21 +808,23 @@ TEST_CASE_METHOD(CApiTest, "CAPI silkworm_execute_blocks_perpetual multiple bloc
 }
 
 TEST_CASE_METHOD(CApiTest, "CAPI silkworm_add_snapshot", "[silkworm][capi]") {
-    snapshot_test::SampleHeaderSnapshotFile valid_header_snapshot{tmp_dir.path()};
-    snapshot_test::SampleHeaderSnapshotPath header_snapshot_path{valid_header_snapshot.path()};
-    snapshot_test::SampleBodySnapshotFile valid_body_snapshot{tmp_dir.path()};
-    snapshot_test::SampleBodySnapshotPath body_snapshot_path{valid_body_snapshot.path()};
-    snapshot_test::SampleTransactionSnapshotFile valid_tx_snapshot{tmp_dir.path()};
-    snapshot_test::SampleTransactionSnapshotPath tx_snapshot_path{valid_tx_snapshot.path()};
+    snapshot_test::SampleHeaderSnapshotFile header_snapshot_file{tmp_dir.path()};
+    auto& header_snapshot_path = header_snapshot_file.path();
+    snapshot_test::SampleBodySnapshotFile body_snapshot_file{tmp_dir.path()};
+    auto& body_snapshot_path = body_snapshot_file.path();
+    snapshot_test::SampleTransactionSnapshotFile tx_snapshot_file{tmp_dir.path()};
+    auto& tx_snapshot_path = tx_snapshot_file.path();
 
     auto header_index_builder = snapshots::HeaderIndex::make(header_snapshot_path);
+    header_index_builder.set_base_data_id(header_snapshot_file.block_num_range().start);
```

Comment:
Adding the public method `IndexBuilder::set_base_data_id` to handle custom start point only for tests is not ideal, but it's necessary to support our sample snapshots used in tests.
A better overall solution would be:
- introduce some subcommand in `snapshots` command-line utility to generate (short) sample snapshots
- use some other command-line utility (maybe new `embed_binary`?) to generate C++ code for sample snapshots
- regenerate sample snapshots without custom start point (i.e. add initial empty blocks)
- remove `IndexBuilder::set_base_data_id` as it becomes unnecessary

I'll open a dedicated issue on this.